testing/internal/e2e/tests/base/paginate_auth_token_test.go

```diff
@@ -0,0 +1,242 @@
+	require.NoError(t, err)
+
+	var returnedIds []string
+	// Ignore the precreated auth token, which will appear at the end
+	for _, authtoken := range initialAuthTokens.Items[:len(initialAuthTokens.Items)-numPrecreatedAuthTokens] {
+		returnedIds = append(returnedIds, authtoken.Id)
+	}
+
+	require.Len(t, initialAuthTokens.Items, c.MaxPageSize+1)
+	assert.Empty(t, cmp.Diff(returnedIds, authTokenIds, cmpopts.SortSlices(func(i, j string) bool { return i < j })))
+	assert.Empty(t, initialAuthTokens.ResponseType)
+	assert.Empty(t, initialAuthTokens.RemovedIds)
+	assert.Empty(t, initialAuthTokens.ListToken)
+
+	// Create a new auth token and destroy one of the other auth tokens
+	newAuthTokenId := createAuthToken(t, ctx, client, amId, "testuser", password)
+	output = e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs(
+			"auth-tokens", "delete",
+			"-id", initialAuthTokens.Items[0].Id,
+		),
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+
+	// List again, should have the new auth token but not the deleted auth token
+	output = e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs(
+			"auth-tokens", "list",
+			"-recursive",
+			"-format=json",
+		),
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+
+	var newAuthTokens authtokens.AuthTokenListResult
+	err = json.Unmarshal(output.Stdout, &newAuthTokens)
+	require.NoError(t, err)
+
+	require.Len(t, newAuthTokens.Items, c.MaxPageSize+1)
+	// The first item should be the most recently created, which
+	// should be our new auth token
+	firstItem := newAuthTokens.Items[0]
+	assert.Equal(t, newAuthTokenId, firstItem.Id)
+	assert.Empty(t, newAuthTokens.ResponseType)
+	assert.Empty(t, newAuthTokens.RemovedIds)
+	assert.Empty(t, newAuthTokens.ListToken)
+	// Ensure the deleted auth token isn't returned
+	for _, authtoken := range newAuthTokens.Items {
+		assert.NotEqual(t, authtoken.Id, initialAuthTokens.Items[0].Id)
+	}
+}
+
+// TestApiPaginateAuthTokens asserts that the API automatically paginates to retrieve
+// all auth tokens in a single invocation.
+func TestApiPaginateAuthTokens(t *testing.T) {
+	e2e.MaybeSkipTest(t)
+	c, err := loadTestConfig()
+	require.NoError(t, err)
+
+	client, err := boundary.NewApiClient()
+	require.NoError(t, err)
+	adminToken := client.Token()
+	ctx := context.Background()
+	sClient := scopes.NewClient(client)
+	amClient := authmethods.NewClient(client)
+	atClient := authtokens.NewClient(client)
+	newOrgId := boundary.CreateNewOrgApi(t, ctx, client)
+	userId := boundary.CreateNewUserApi(t, ctx, client, newOrgId)
+	amId := boundary.CreateNewAuthMethodApi(t, ctx, client, newOrgId)
+	accId, password := boundary.CreateNewAccountCli(t, ctx, amId, "testuser")
+	boundary.SetAccountToUserCli(t, ctx, userId, accId)
+	t.Cleanup(func() {
+		ctx := context.Background()
+		client.SetToken(adminToken)
+		_, err := amClient.Delete(ctx, amId)
+		require.NoError(t, err)
+		_, err = sClient.Delete(ctx, newOrgId)
+		require.NoError(t, err)
+	})
```

Comment:
Should we have the org cleanup occur just after org creation just in case something happens to fail beforehand for whatever reason?